src/vcpkg/tools.cpp

```diff
@@ -638,13 +638,38 @@ gsutil version: 4.58
 
     struct ToolCacheImpl final : ToolCache
     {
-        ToolCache::RequireExactVersions abiToolVersionHandling;
+        RequireExactVersions abiToolVersionHandling;
+        vcpkg::Cache<std::string, Path> system_cache;
         vcpkg::Cache<std::string, Path> path_only_cache;
         vcpkg::Cache<std::string, PathAndVersion> path_version_cache;
 
-        ToolCacheImpl(ToolCache::RequireExactVersions abiToolVersionHandling)
-            : abiToolVersionHandling(abiToolVersionHandling)
+        ToolCacheImpl(RequireExactVersions abiToolVersionHandling) : abiToolVersionHandling(abiToolVersionHandling) { }
+
+        virtual const Path& get_tool_path_from_system(const Filesystem& fs, const std::string& tool) const override
         {
+            return system_cache.get_lazy(tool, [&] {
+                if (tool == Tools::TAR)
+                {
+                    auto tars = fs.find_from_PATH("tar");
+                    if (tars.empty())
+                    {
+                        Checks::exit_with_message(VCPKG_LINE_INFO,
+#if defined(_WIN32)
+                                                  "Could not find tar; the action you tried to take assumes Windows 10 "
+                                                  "or later which are bundled with tar.exe."
+#else  // ^^^ _WIN32 // !_WIN32 vvv
+                                                  "Could not find tar; please install it from your system package "
+                                                  "manager."
+#endif // ^^^ !_WIN32
+
+                        );
+                    }
+
+                    return tars[0];
```

Comment:
```suggestion
                    return std::move(tars[0]);
```

Reply:
I think this is one of the cases that would get implicitly moved but at least this one is never eligible for NRVO, so I applied this.